src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -2174,6 +2178,44 @@ MavlinkReceiver::handle_message_hil_sensor(mavlink_message_t *msg)
 	}
 }
 
+void
+MavlinkReceiver::handle_message_gps_input(mavlink_message_t *msg)
+{
+	mavlink_gps_input_t gps_input;
+	mavlink_msg_gps_input_decode(msg, &gps_input);
+
+	uint64_t timestamp = hrt_absolute_time();
+
+	vehicle_gps_position_s gps = {};
+
+	gps.timestamp_time_relative = 0;
+	gps.time_utc_usec = gps_input.time_usec;
+
+	gps.timestamp = timestamp;
+	gps.lat = gps_input.lat;
+	gps.lon = gps_input.lon;
+	gps.alt = gps_input.alt;
+	gps.eph = (float)gps_input.horiz_accuracy * 1e-2f; // from cm to m
+	gps.epv = (float)gps_input.vert_accuracy * 1e-2f; // from cm to m
```

Comment:
The spec says this is all meters, not cm.

https://mavlink.io/en/messages/common.html#GPS_INPUT